src/app/ReadHandler.cpp

```diff
@@ -52,11 +52,13 @@ uint16_t ReadHandler::GetPublisherSelectedIntervalLimit()
 #endif
 }
 
-ReadHandler::ReadHandler(ManagementCallback & apCallback, Messaging::ExchangeContext * apExchangeContext,
+ReadHandler::ReadHandler(InteractionModelEngine * apImEngine, ManagementCallback & apCallback, Messaging::ExchangeContext * apExchangeContext,
```

Comment:
Given that the ManagementCallback _is_ already the IM engine, this seems like a public API change plus memory bloat which could probably be avoided by adding a way to get the IM engine to ManagementCallback, no?

More generally, it's not clear to me why ManagementCallback needs to exists at all if we have an IM engine pointer....

If we're going to make any sort of incompatible changes, we should design the API to actually make sense so we don't keep making them.

Reply:
Added a getter in ManagementCallback to get the IM Engine and keep the constructor unchanged.